src/App.tsx

```diff
@@ -1,26 +1,39 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect } from 'react';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { todosSlice } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const hasTodos = useAppSelector(state => state.todos.length > 0);
+  const hasSelectedTodo = useAppSelector(state => state.currentTodo !== null);
+  const dispatch = useAppDispatch();
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    getTodos().then(todosFromServer =>
+      dispatch(todosSlice.actions.set(todosFromServer)),
```

Comment:
Ensure that the `todosSlice.actions.set` action is correctly defined in the `todosSlice`. If this action does not exist, it will cause an error when trying to dispatch it.